client/src/components/classes/CampaignForm.js

```diff
@@ -1,215 +1,108 @@
-        wrong_answer:
-        right_answer:
-      },
-      {
-        id:
-        question:
-        wrong_answer:
-        right_answer:
-      }
-    ]
-  }
-]
-
-Refreshr =
-Questions =
-
-Refreshrs= {
-  refreshr_1_aka_day2: {
-    id: 3431
-    name
-    date 2/21/2019
-    reviewtext
-
-    questions: [
-      question_1: {
-        id
-        question
-        wrong,
-        right
-      },
-      question_2: {
-        id
-        question
-        wrong,
-        right
-      },
-      question_3: {
-        id
-        question
-        wrong,
-        right
-      }
-    ]
-  },
-  refreshr_1_aka_week2: {
-    id: 1341
-    name
-    date
-    reviewtext
-
-    questions: [
-      question_1: {
-        id
-        question
-        wrong,
-        right
-      },
-      question_2: {
-        id
-        question
-        wrong,
-        right
-      },
-      question_3: {
-        id
-        question
-        wrong,
-        right
-      }
-    ]
-  },
-}
-
-1. Structure of Refreshr Tree
-2. Adding name to each Refreshr
-3. Each Refreshr sends out the same exact questions
-4. Recipients Operation
-
-*/
 function CampaignForm(props) {
-  console.log('campaign form props', props);
-  
-  const addDate = (date, id) => {
-    console.log(props.refreshrs[0], typeof id)
-    let date2 = new Date(date);
-    console.log(date2, typeof date2)
-    const refreshr = props.refreshrs.filter(r => r.refreshr_id === id);
-    console.log(refreshr);
-    refreshr.date = date2;
-    if (refreshr.length === 1) {
+  const [refreshrs, setRefreshrs] = useState([]);
+  const { classes } = props;
+  let today = new Date(); // to set default for date inputs
+  today = dateMapper(today);
 
-    }
-    console.log(`after: ${refreshr.date}`);
-    // console.log(date);
+  // fetch class refreshrs on mount
+  useEffect(() => {
+    getRefreshrs();
+  }, []);
 
+  /* putting the axios request here for now just to design the page. it may make more sense to make
+  it in the parent component and then pass down props to the children components */
+  const getRefreshrs = async () => {
+    try {
+      /* this should fetch the class's refreshrs from /refreshrs/classes/:classId,
+        but the endpoint is not live yet so I'm using this for testing */
+      const res = await axios.get('https://refreshr.herokuapp.com/refreshrs');
+      setRefreshrs(res.data.refreshrs.slice(0, 6));
+    } catch (err) {
+      console.log(err);
+    }
+  };
 
+  // got to be a cleaner way to do this, but it works for now
+  function dateMapper(date) {
+    const month =
+      date.getMonth() < 9 ? `0${date.getMonth() + 1}` : date.getMonth() + 1;
+    const day = date.getDate() < 10 ? `0${date.getDate()}` : date.getDate();
+    const year = date.getFullYear();
+    return `${year}-${month}-${day}`;
   }
 
+  // updates the refreshr's date when date input is changed
+  const setDate = (date, id) => {
+    const [refreshr] = refreshrs.filter(r => r.id === id);
```

Comment:
Cool, you got the filter working?

Reply:
Yeah, it's a different axios call for just the refreshr list, so it was a little easier to filter